### ISSUE TITLE
`nixops deploy` fails with: "/root/.nix-profile/etc/profile.d/nix.sh: No such file or directory"

### ISSUE THREAD

**Issue by koslambrou** [NixOS/nixops#1377]
Here are my input files:

trivial.nix
```
{
  resources.sshKeyPairs.ssh-key = {};

  webserver = { pkgs, ... }: {
    services.openssh.enable = true;

    deployment.targetEnv = "digitalOcean";
    deployment.digitalOcean.enableIpv6 = false;
    deployment.digitalOcean.region = "nyc3";
    deployment.digitalOcean.size = "s-1vcpu-1gb";
  };
}
```

trivial-digital-ocean.nix
```
{
  network.description = "Web server";

  webserver =
    { config, pkgs, ... }:
    { services.httpd.enable = true;
      services.httpd.adminAddr = "alice@example.org";
      services.httpd.documentRoot = "${pkgs.valgrind.doc}/share/doc/valgrind/html";
      networking.firewall.allowedTCPPorts = [ 80 ];
    };
}
```

I created a deployment with: `nixops create trivial.nix trivial-digital-ocean.nix -d trivial` and then deployed with `DIGITAL_OCEAN_AUTH_KEY=<YOURKEY> nixops deploy -d trivial`

Output:
```
webserver> creating droplet ...[in-progress] [in-progress] [in-progress] [in-progress] [in-progress] [in-progress] [in-progress] [in-progress] [in-progress] [completed] 165.227.103.81
webserver> waiting for SSH.......................
webserver> running nixos-infectWarning: Permanently added '165.227.103.81' (ECDSA) to the list of known hosts.

webserver> + makeConf
webserver> + mkdir -p /etc/nixos
webserver> + local 'IFS=
webserver> '
webserver> + keys=($(grep -vE '^[[:space:]]*(#|$)' /root/.ssh/authorized_keys))
webserver> ++ grep -vE '^[[:space:]]*(#|$)' /root/.ssh/authorized_keys
webserver> + cat
webserver> ++ hostname
webserver> ++ for key in '${keys[@]}'
webserver> ++ echo -n '
webserver>     "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIAglUkQ7VzRAULTQcItRVdL5D7MrSjNIKGngS6Us6CBk NixOps auto-generated key"'
webserver> + cat
webserver> + local 'IFS=
webserver> '
webserver> + ens3_ip4s=($(ip address show dev eth0 | grep 'inet '          | sed -r 's|.*inet ([0-9.]+)/([0-9]+).*|{ address="\1"; prefixLength=\2; }|'))
webserver> ++ sed -r 's|.*inet ([0-9.]+)/([0-9]+).*|{ address="\1"; prefixLength=\2; }|'
webserver> ++ grep 'inet '
webserver> ++ ip address show dev eth0
webserver> + ens3_ip6s=($(ip address show dev eth0 | grep 'inet6 .*global' | sed -r 's|.*inet6 ([0-9a-f:]+)/([0-9]+).*|{ address="\1"; prefixLength=\2; }|'))
webserver> ++ sed -r 's|.*inet6 ([0-9a-f:]+)/([0-9]+).*|{ address="\1"; prefixLength=\2; }|'
webserver> ++ grep 'inet6 .*global'
webserver> ++ ip address show dev eth0
webserver> + ens4_ip4s=($(ip address show dev eth1 | grep 'inet '          | sed -r 's|.*inet ([0-9.]+)/([0-9]+).*|{ address="\1"; prefixLength=\2; }|'))
webserver> ++ sed -r 's|.*inet ([0-9.]+)/([0-9]+).*|{ address="\1"; prefixLength=\2; }|'
webserver> ++ grep 'inet '
webserver> ++ ip address show dev eth1
webserver> Device "eth1" does not exist.
webserver> + ens4_ip6s=($(ip address show dev eth1 | grep 'inet6 .*global' | sed -r 's|.*inet6 ([0-9a-f:]+)/([0-9]+).*|{ address="\1"; prefixLength=\2; }|'))
webserver> ++ grep 'inet6 .*global'
webserver> ++ sed -r 's|.*inet6 ([0-9a-f:]+)/([0-9]+).*|{ address="\1"; prefixLength=\2; }|'
webserver> ++ ip address show dev eth1
webserver> Device "eth1" does not exist.
webserver> + gateway=($(ip route show dev eth0 | grep default | sed -r 's|default via ([0-9.]+).*|\1|'))
webserver> ++ sed -r 's|default via ([0-9.]+).*|\1|'
webserver> ++ grep default
webserver> ++ ip route show dev eth0
webserver> + gateway6=($(ip -6 route show dev eth0 | grep default | sed -r 's|default via ([0-9a-f:]+).*|\1|'))
webserver> ++ grep default
webserver> ++ sed -r 's|default via ([0-9a-f:]+).*|\1|'
webserver> ++ ip -6 route show dev eth0
webserver> + ether0=($(ip address show dev eth0 | grep link/ether | sed -r 's|.*link/ether ([0-9a-f:]+) .*|\1|'))
webserver> ++ sed -r 's|.*link/ether ([0-9a-f:]+) .*|\1|'
webserver> ++ grep link/ether
webserver> ++ ip address show dev eth0
webserver> + ether1=($(ip address show dev eth1 | grep link/ether | sed -r 's|.*link/ether ([0-9a-f:]+) .*|\1|'))
webserver> ++ grep link/ether
webserver> ++ ip address show dev eth1
webserver> ++ sed -r 's|.*link/ether ([0-9a-f:]+) .*|\1|'
webserver> Device "eth1" does not exist.
webserver> + nameservers=($(grep ^nameserver /etc/resolv.conf | cut -f2 -d' '))
webserver> ++ cut -f2 '-d '
webserver> ++ grep '^nameserver' /etc/resolv.conf
webserver> + cat
webserver> ++ for a in '${nameservers[@]}'
webserver> ++ echo -n '
webserver>       "67.207.67.2"'
webserver> ++ for a in '${nameservers[@]}'
webserver> ++ echo -n '
webserver>       "67.207.67.3"'
webserver> ++ for a in '${ens3_ip4s[@]}'
webserver> ++ echo -n '
webserver>           { address="165.227.103.81"; prefixLength=20; }'
webserver> ++ for a in '${ens3_ip4s[@]}'
webserver> ++ echo -n '
webserver>           { address="10.17.0.5"; prefixLength=16; }'
webserver> ++ type -t customConfig
webserver> + [[ '' == \f\u\n\c\t\i\o\n ]]
webserver> + makeSwap
webserver> ++ mktemp
webserver> + swapFile=/tmp/tmp.gecRvwt0cE
webserver> + dd if=/dev/zero of=/tmp/tmp.gecRvwt0cE bs=1M count=1024
webserver> 1024+0 records in
webserver> 1024+0 records out
webserver> 1073741824 bytes (1.1 GB, 1.0 GiB) copied, 2.33297 s, 460 MB/s
webserver> + chmod 0600 /tmp/tmp.gecRvwt0cE
webserver> + mkswap /tmp/tmp.gecRvwt0cE
webserver> Setting up swapspace version 1, size = 1024 MiB (1073737728 bytes)
webserver> no label, UUID=7d9d0c6d-9c96-4c53-840d-4dda483e44fc
webserver> + swapon /tmp/tmp.gecRvwt0cE
webserver> + which dnf
webserver> + export USER=root
webserver> + USER=root
webserver> + export HOME=/root
webserver> + HOME=/root
webserver> + groupadd -r nixbld -g 30000
webserver> + seq 1 10
webserver> ++ which nologin
webserver> + xargs '-I{}' useradd -c 'Nix build user {}' -d /var/empty -g nixbld -G nixbld -M -N -r -s /usr/sbin/nologin 'nixbld{}'
webserver> + sh
webserver> + curl https://nixos.org/nix/install
webserver>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
webserver>                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
webserver> + source /root/.nix-profile/etc/profile.d/nix.sh
webserver> bash: line 159: /root/.nix-profile/etc/profile.d/nix.sh: No such file or directory
```

The reason is that `https://nixos.org/nix/install` returns the HTTP code `301 Moved Temporarly`, so the `-L` flag should be added to the `curl` command.

Seems like its fixed in a recent [pull request](https://github.com/elitak/nixos-infect/pull/55) of `nixos-infect`, so an update of the library is necessary.

**Comment by kshaa** [comment]
I created a PR for this, just needs to be merged - https://github.com/NixOS/nixops/pull/1379

**Comment by kshaa** [comment]
While the PR isn't merged yet, it's possible to work around it by importing NixOps from my repo.

a. use fetchTarball with the following attributes:
- url = https://github.com/kshaa/nixops/archive/fb6d4665e8efd858a215bbaaf079ec3f5ebc49b8.tar.gz;
- sha256 = "1rkk31ka4w5dklcwvy3s73pi6yjymijcnmjq4n3wkc1rjs5ck7y2";

b. use [niv](https://github.com/nmattia/niv)
- `niv add kshaa/nixops -b release-1.7`

And remember to pass the attribute `nixpkgs` with version `19.03`, I tried `19.09` and `20.03` but they have other issues. For more info on building NixOps I used [this Discourse post](https://discourse.nixos.org/t/what-am-i-doing-wrong-here/2517/9).
